### ISSUE TITLE
Shell login/logout title don't follow databinding

### ISSUE THREAD

**Issue by ncarandini** [ncarandini/AuthDemoOld#1]
The shell tab item:
```
<Tab Title="{Binding LogInOutTitle}" Icon="{Binding LogInOutIconName}">
    <ShellContent ContentTemplate="{DataTemplate views:LoginPage}" />
</Tab>
```
has an issue with the Title, that don't update from "Login" to "Logout" when the user logs in.
Oddly enought, the Icon works fine.

**Comment by biapar** [comment]
Maybe an error into ViewModelBase.cs. The flag IsLogged is always false.

**Comment by ncarandini** [comment]
@biapar there is no `IsLogged` variable in the `ViewModelBase.cs`.
I have:
```
public bool IsUserLogged => AuthenticationService.UserContext?.IsLoggedOn ?? false;
```
Moreover, on the AppShellViewModel I don't use it as I have this method tied to a `AuthenticationStateChanged` event:
```
private void OnAuthenticationStateChanged(object sender, EventArgs e)
{
    LogInOutTitle = !AuthenticationService.UserContext?.IsLoggedOn ?? true ? "Login" : "Logout";

    OnPropertyChanged(nameof(LogInOutTitle));
    OnPropertyChanged(nameof(LogInOutIconName));
}
```
and on the AppShell I have:
```
<Tab Title="{Binding LogInOutTitle}" Icon="{Binding LogInOutIconName}">
    <ShellContent ContentTemplate="{DataTemplate views:LoginPage}" />
</Tab>
```
Finally, when running the app with the mocked authentication service, you can see that the icon change but the text don't:
![image](https://user-images.githubusercontent.com/139274/82948901-f7416d80-9fa2-11ea-803d-79cf9cb52e73.png)
To me this smell as a Xamarin Forms Shell bug.

**Comment by biapar** [comment]
In this class ViewModelBase : BaseViewModel

```
 async Task UserLogin()
        {
            if (!IsLogging)
            {
                IsLogging = true;
                try
                {
                    // TODO: Analytics.TrackEvent("UserLogin");
                    await AuthenticationService.SignInAsync();
                }
                catch (Exception)
                {
                }
                finally
                {
                    IsLogging = false;
                }
            }
        }
```

**Comment by biapar** [comment]
Other strange thing is that I'm able to register a new user, login via Facebook but the app don't receive the token.
I see the users into the ADB2C users.

**Comment by ncarandini** [comment]
@biapar read carefully, here I'm using a `IsLogging` variable to mark the ongoing logging operation, not `IsLogged` ! The semantic is completely different (and the name also 😄 ). It's true while logging, then is is set to false.

**Comment by ncarandini** [comment]
For the Facebook thing, you'd better open another issue so we can work on it and track it separately.

**Comment by biapar** [comment]
Ok. I did not see that the icon change in tab bar. I'll check again. My mistake.

**Comment by ncarandini** [comment]
1. Did you see that you missed the name of the variable (IsLogging instead of IsLogged )?
if so, please acknowledge the fact. Otherwise I can't know if I'm still missing something about your comments.
2. The icon changes and you can see it also in the image I've posted. the login icon has the frame hole on the **left side** and the arrow going from **outside left** to **inside right**, while the logout icon has the frame hole on the **right side** and the arrow going from **inside left** to **outside right**. Find the difference!

**Comment by biapar** [comment]
1. I remove the IsLogging = false; from finally branch because I see always the login link also if I logged. But there is a strange think that I see while debugging that cause that variable go to false. In the SignInAsync() function the token is null.
```
finally
            {
                //Why pass AccessToken null if I logged and before it was not null????? -> BP20200527
                await SetAccessTokenAsync(userContext?.AccessToken);
                UserContext = userContext;
                OnAuthenticationStateChanged();
            }

```
![image](https://user-images.githubusercontent.com/661517/83191641-a5c8e800-a134-11ea-9864-2ef3993d0b83.png)

2. I see that. I'm trying to understand why the label don't change.

**Comment by biapar** [comment]
Solved with a workaround. I use the messages. I remove binding on Title.

In AppShellViewModel:
```
private void OnAuthenticationStateChanged(object sender, EventArgs e)
        {
            LogInOutTitle = !(AuthenticationService.UserContext?.IsLoggedOn) ?? true ? "Login" : "Logout";

            MessagingCenter.Send<AppShellViewModel, string>(this, "changeTitle", LogInOutTitle);

            OnPropertyChanged(nameof(LogInOutTitle));
            OnPropertyChanged(nameof(LogInOutIconName));
        }
```


In AppShell
```
MessagingCenter.Subscribe<AppShellViewModel, string>(this, "changeTitle", (sender, arg) =>
             {
                 this.tabLogin.Title = arg;
             });
```

![image](https://user-images.githubusercontent.com/661517/83193576-91d2b580-a137-11ea-9675-82ca43402e55.png)

**Comment by ncarandini** [comment]
> //Why pass AccessToken null if I logged and before it was not null?????

Usually, but I have to further investigate if this is the case, just for security reason. But this has nothing to do with IsLogging that is a flag that is true only while logging, and then has to be set to false, so it's wrong to remove it.

**Comment by biapar** [comment]
I restored the original code. The tab title don't change also if the flags are ok.

**Comment by ncarandini** [comment]
The workaround works, but why using Messaging Center when there is already an event you can subscribe? Hope you don't mind if I'll do it differently.

**Comment by ncarandini** [comment]
> I restored the original code. The tab title don't change also if the flags are ok.

True, I've created an [issue on Xamarin forms](https://github.com/xamarin/Xamarin.Forms/issues/10877) for the bug and a [repro](https://github.com/ncarandini/XFShellBug)
